### PR TITLE
🚧(ui): Adding API service, integrating NGXS

### PR DESCRIPTION
**Pull request by dylan2894**
- Added a service that consumes the request-for-access API. 
- Integrated the NGXS state management with the request-for-access API calls. 
- Finished responsive styling. 
- Optimised HTML.

### CODE REVIEW

**Review by arulagrawal — DISMISSED** [review]

File: libs/client/request-for-access/feature/src/lib/request-for-access.actions.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export class SetAccessStatus {
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/request-for-access.actions.ts:1]: This is fine for now. But consider putting actions in its own folder.

File: libs/client/request-for-access/feature/src/lib/request-for-access.component.ts [hunk 2]
```diff
@@ -1,16 +1,69 @@
...
+    this.isFeatureVisible = true;
+    this.companyID = "";
+    this.graduateID = "";
+  }
+
+  ngOnInit(): void {
+    const actionsArr: SetAccessStatus[] = [];
+
+    //make API call to access status of resources for particular company
+      this.apiService.getResourceStatuses('4', '2').subscribe({
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/request-for-access.component.ts:29]: Technically speaking, this part need to be in a NGXS Store. But I will accept this for now. Has to be changed in the future though.

File: libs/client/request-for-access/feature/src/lib/request-for-access.component.ts [hunk 3]
```diff
@@ -1,16 +1,69 @@
...
+        next: (_res) => {
+          const jsonString = JSON.stringify(_res.response);
+          const response = JSON.parse(jsonString);
+
+          let i=0;
+          while(response.data.status[i] != undefined) {
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/request-for-access.component.ts:35]: While loops are always scary. Can we not do this differently?

File: libs/client/request-for-access/feature/src/lib/request-for-access.module.ts [hunk 4]
```diff
@@ -3,12 +3,23 @@ import { NgModule } from '@angular/core';
...
 
 @NgModule({
-  imports: [CommonModule, MatDividerModule, FlexLayoutModule],
+  imports: [
+    CommonModule,
+    MatDividerModule,
+    FlexLayoutModule,
+    HttpClientModule,
+    MatIconModule,
+    NgxsModule.forRoot([
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/request-for-access.module.ts:18]: This should not be forRoot. Should be forFeature. If you want to add the store to the "global" scope. Then the store should be configured in the shell's module.

File: libs/client/request-for-access/feature/src/lib/request-for-access.state.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { SetAccessStatus } from './request-for-access.actions';
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/request-for-access.state.ts:1]: Convention is to put states in a stores folder.

File: libs/client/request-for-access/feature/src/lib/services/request-for-access.service.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { ajax, AjaxResponse } from 'rxjs/ajax';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RequestForAccessService {
+    getResourceStatuses(companyID: string, graduateID: string): Observable<AjaxResponse<JSON>> {
+      return ajax({
```
Comment (arneschreuder) [inline libs/client/request-for-access/feature/src/lib/services/request-for-access.service.ts:10]: Why Ajax? Angular has a built in httpClient that makes this SO much easier. Please change out.

**Review by arneschreuder — CHANGES_REQUESTED** [review]
Not a bad PR. Some convention stuff that should be followed.

**Comment by dylan2894** [comment]
@arneschreuder Should the actions folder and stores folder be nested inside the lib folder? (as it is now with my new commits)

File: libs/client/request-for-access/feature/src/lib/services/request-for-access.service.ts (hunk 6 above)
Comment (arneschreuder) [reply libs/client/request-for-access/feature/src/lib/services/request-for-access.service.ts]: How is this resolved?

**Review by arneschreuder — APPROVED** [review]
Good stuff. Thanks

**Comment by arneschreuder** [comment]
Go ahead and merge!

**Pull request merged**